File: Tema5/Objetos/src/model/Persona.java
package model;

import java.util.ArrayList;

public class Persona {
    private String dni, nombre, apellido;
    private ArrayList <Persona> listaPersona;

    public Persona(String dni, String nombre, String apellido){
        listaPersona=new ArrayList<>();
        this.dni=dni;
        this.nombre=nombre;
        this.apellido=apellido;
    }

    public void mostrarDatos(){
        System.out.println("El dni es "+this.dni);
        System.out.println("El nombre es "+this.nombre);
        System.out.println("El apellido es "+this.apellido);

    }
}

File: Kotlin/TrabajadoresHerencia/src/model/Persona.kt
package model

open class Persona(var nombre: String, var apellido: String, var dni: String) {
    var telefono: Int? = null
    var correoE: String? = null

    constructor(nombre: String, apellido: String, dni: String, telefono: Int?, correo: String?) : this(nombre, apellido, dni) {
        this.telefono = telefono
        this.correoE = correo
    }

    open fun mostrarDatos() {
        println("Nombre: $nombre")
        println("Apellido: $apellido")
        println("DNI: $dni")
        println("Teléfono: ${telefono ?: "no especificado"}")
        println("Correo: ${correoE ?: "no especificado"}")
    }
}

